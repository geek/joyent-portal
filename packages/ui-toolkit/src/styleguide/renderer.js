import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
