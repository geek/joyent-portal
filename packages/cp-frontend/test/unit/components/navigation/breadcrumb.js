/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Breadcrumb from '@components/navigation/breadcrumb';
import { Router } from '../../mocks';

it('renders <Breadcrumb /> without throwing', () => {
  const tree = renderer
    .create(
      <Router>
        <Breadcrumb />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
