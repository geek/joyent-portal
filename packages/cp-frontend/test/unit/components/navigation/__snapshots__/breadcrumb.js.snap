// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Breadcrumb /> without throwing 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.c0 {
  border-bottom: solid 0.0625rem;
}

@media only screen and (min-width:48em) {
  .c1 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c1 {
    width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c1 {
    width: 76rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      name="breadcrum"
    />
  </div>
</div>
`;
